cpp/src/fil/internal.cuh

```diff
@@ -447,28 +451,28 @@ struct cat_sets_device_owner {
   // arrays from each node ID are concatenated first, then from all categories
   rmm::device_uvector<uint8_t> bits;
   // largest matching category in the model, per feature ID
-  rmm::device_uvector<int> max_matching;
+  rmm::device_uvector<float> fid_cats;
 
   categorical_sets accessor() const
   {
     return {
-      .bits              = bits.data(),
-      .max_matching      = max_matching.data(),
-      .bits_size         = bits.size(),
-      .max_matching_size = max_matching.size(),
+      .bits          = bits.data(),
+      .fid_cats      = fid_cats.data(),
+      .bits_size     = bits.size(),
+      .fid_cats_size = fid_cats.size(),
     };
   }
-  cat_sets_device_owner(cudaStream_t stream) : bits(0, stream), max_matching(0, stream) {}
+  cat_sets_device_owner(cudaStream_t stream) : bits(0, stream), fid_cats(0, stream) {}
   cat_sets_device_owner(categorical_sets cat_sets, cudaStream_t stream)
-    : bits(cat_sets.bits_size, stream), max_matching(cat_sets.max_matching_size, stream)
+    : bits(cat_sets.bits_size, stream), fid_cats(cat_sets.fid_cats_size, stream)
   {
-    ASSERT(bits.size() <= static_cast<std::size_t>(INT_MAX) + 1ull,
+    ASSERT(bits.size() <= static_cast<std::size_t>(INT_MAX) + std::size_t(1),
```

Comment:
Out of curiosity, why do we initialize one `size_t` from an int here but use `static_cast` for the other?